build_defs.bzl

```diff
@@ -68,10 +68,7 @@ def pybind_extension(
         linkopts = linkopts + select({
             "@platforms//os:osx": ["-undefined", "dynamic_lookup"],
             Label("@pybind11//:msvc_compiler"): [],
-            "//conditions:default": [
-                "-Wl,-Bsymbolic",
-                "-Wl,-z,undefs",
```

Comment:
Do you still need `"-Wl,-z,undefs"`?

Reply:
I don't. I am just reverting the PR to restore the previous working state.

Reply:
@jiawen Kind reminder

Reply:
My apologies. I've been slammed with other stuff but I'll take a look at it this weekend.

Reply:
Sorry this took so long. I finally skimmed what `-z undefs` means (thanks @MaskRay !) - yeesh this is a rabbit hole I did not expect to go into.

For some reason, I misread your PR - `-Wl,-Bsymbolic` was pre-existing. Approving now.